### ISSUE TITLE
android issues

### ISSUE THREAD

**Issue by weinixuehao** [fengjian0106/hed-tutorial-for-document-scanning#8]
i had been ported to android, but precise is less than ios. 

the android underlying code is the same as ios since i using c++.

why?

tensorflow 1.7 android ios both

opencv ios the same as yours

android use the latest version of opencv, but code is the same as ios, since core code is same(c++)

**Comment by fengjian0106** [comment]
@weinixuehao  精度的差异有多大，能截个图或录屏出来看一下吗？可以仿照 ios_demo 那样，把中间过程的 debug image 也实时的显示出来。
另外，你是否可以初步判断一下，出现的精度差异，是来源于HED边缘检测部分，还是来源于OpenCV实现的找点算法部分？
在我自己的app里面，部署到Android上的时候，HED部分和找点算法部分，在精确度上是没有遇到明显的差异的，唯一遇到的问题是 batch norm 在 32bit 的 Android 上运行速度很慢，所以我们在 Android 上做了一些降级处理，使用的是 VGG style 的 HED，而不是像 iOS 那样使用的 mobilenet style HED，这份代码对应的博客里面有描述这个情况

**Comment by weinixuehao** [comment]
我发现一个现象，imgfile -> bitmap -> mat 精度有误差。如果是直接file(imread) -> mat 和ios精度是一样的。

**Comment by weinixuehao** [comment]
@fengjian0106 你有微信或者qq么？

**Comment by weinixuehao** [comment]
@fengjian0106 我怀疑是bitmap to mat有像素丢失，我实时看过bitmap显示和Ios显示清晰度比不上ios。

**Comment by weinixuehao** [comment]
1，我太粗心了，两个图片一个横屏测，一个竖屏的测。不好意思，打扰了。

**Comment by weinixuehao** [comment]
bitmap to mat 按照opencv的源码处理颜色空间会变。
bitmap bgr channel
opencv mat 是rgb channel

**Comment by fengjian0106** [comment]
查到问题了就行
